src/btree/bt_curprev.c

```diff
@@ -852,21 +859,45 @@ __wt_btcur_prev(WT_CURSOR_BTREE *cbt, bool truncating)
         /*
          * If we saw a lot of deleted records on this page, or we went all the way through a page
          * and only saw deleted records, try to evict the page when we release it. Otherwise
-         * repeatedly deleting from the beginning of a tree can have quadratic performance. Take
-         * care not to force eviction of pages that are genuinely empty, in new trees.
-         *
-         * A visible stop timestamp could have been treated as a tombstone and accounted in the
-         * deleted count. Such a page might not have any new updates and be clean, but could benefit
-         * from reconciliation getting rid of the obsolete content. Hence mark the page dirty to
-         * force it through reconciliation.
+         * repeatedly searching from the beginning of a tree can have quadratic performance. Take
+         * care not to force eviction of genuinely empty pages, in new trees.
          */
-        if (page != NULL &&
-          (cbt->page_deleted_count > WT_BTREE_DELETE_THRESHOLD ||
-            (newpage && cbt->page_deleted_count > 0))) {
-            WT_ERR(__wt_page_dirty_and_evict_soon(session, cbt->ref));
-            WT_STAT_CONN_INCR(session, cache_eviction_force_delete);
+        if (page != NULL) {
```

Comment:
Can we extract this logic in a separate function and share them between prev and next?

Reply:
Done.